src/Phinx/Config/Config.php

```diff
@@ -294,6 +306,58 @@ public function getTemplateClass()
      }
 
     /**
+     * Get the version order.
+     *
+     * @return string
+     */
+    public function getVersionOrder()
+    {
+        return $this->values['version_order'];
+    }
+
+    /**
+     * Set the version order.
+     * @param string $versionOrder The Version Order to set.
+     */
+    public function setVersionOrder($versionOrder = null)
+    {
+        if (!isset($versionOrder)) {
+            $versionOrder = isset ( $this->values['version_order'] ) ? $this->values['version_order'] : null;
+        }
+
+        if (!isset($versionOrder)) {
+            // if the configuration value is missing we use the default version order (creation time)
+            $this->values['version_order'] = self::VERSION_ORDER_CREATION_TIME;
```

Comment:
or it actually sounds like you're suggesting something like:

```php
switch($this->getConfig()->getVersionOrder()) {
    case \Phinx\Config\Config::VERSION_ORDER_START_TIME:
        // handle start time version order
        break;
    case \Phinx\Config\Config::VERSION_ORDER_CREATION_TIME:
    default:
        // handle creation time version order
}
```

The problem I see here is that an invalid version order (eg. a typo like `starttime`) would be invisibly handled as a creation-time, which could lead to unexpected behaviour (since phinx would rollback the "invalid" commit, at least in the eyes of the user). I believe we should guard against this, if not on the Config getter/setter, then at least in the switch's `default` clause, by raising an exception. If we do so, the only thing left to do is to return the default version order (creation-time) in the getter if none was set in the configuration file.